resources/lib/modules/pulseaudio.py

```diff
@@ -0,0 +1,228 @@
+    # # Menu functions
+    # ###################################################################
+
+    @log.log_function()
+    def set_fallback_sink(self, listItem=None):
+        if listItem is None:
+            listItem = oe.winOeMain.getControl(oe.listObject['palist']).getSelectedItem()
+        if listItem is None:
+            return
+
+        sink = listItem.getProperty('entry')
+        dbus_pulseaudio.core_set_fallback_sink(sink)
+
+    @log.log_function()
+    def change_profile(self, listItem=None):
+        if listItem is None:
+            listItem = oe.winOeMain.getControl(oe.listObject['palist']).getSelectedItem()
+        if listItem is None:
+            return
+
+        card = listItem.getProperty('Card')
+        profiles = dbus_pulseaudio.card_get_property(card, 'Profiles')
+        activeProfile = dbus_pulseaudio.card_get_property(card, 'ActiveProfile')
+
+        items = []
+
+        # we only want to list the available profiles
+        profiles = [profile for profile in profiles if dbus_pulseaudio.profile_get_property(profile, 'Available') == 1]
+        items = [dbus_pulseaudio.profile_get_property(profile, 'Description') for profile in profiles]
+
+        try:
+            active = profiles.index(activeProfile)
+        except ValueError:
+            active = 0
+
+        select_window = xbmcgui.Dialog()
+        title = oe._(32502)
+        result = select_window.select(title, items, preselect=active)
+        if result >= 0:
+            dbus_pulseaudio.card_set_active_profile(card, profiles[result])
+
+    # ###################################################################
+    # # Pulseaudio GUI
+    # ###################################################################
+
+    @log.log_function()
+    def clear_list(self):
+        remove = [entry for entry in self.listItems]
+        for entry in remove:
+            del self.listItems[entry]
+
+    @log.log_function()
+    def menu_connections(self, focusItem=None):
+        if not hasattr(oe, 'winOeMain'):
+            return 0
+        if not oe.winOeMain.visible:
+            return 0
+        if not dbus_pulseaudio.system_has_pulseaudio():
+            oe.winOeMain.getControl(1601).setLabel(oe._(32503))
+            self.clear_list()
+            oe.winOeMain.getControl(int(oe.listObject['palist'])).reset()
+            oe.dbg_log('pulseaudio::menu_connections', 'exit_function (PA Disabled)', oe.LOGDEBUG)
+            return
+        oe.winOeMain.getControl(1601).setLabel(oe._(32504))
+        dictProperties = {}
+
+        # type 1=int, 2=string
+
+        properties = [
+            {
+                'type': 2,
+                'value': 'Driver',
+            },
+            {
+                'type': 2,
+                'value': 'Card',
+            },
+        ]
+
+        rebuildList = 0
+        self.dbusDevices = self.get_sinks()
+        for dbusDevice in self.dbusDevices:
```

Comment:
What is the purpose if this loop? From the looks of it this block is executed once if `self.dbusDevices` is not empty so it can be replaced with `if self.dbusDevices:` condition. Or there is something I'm missing? I saw the same code in bluetooth module and wonder why it's implemented this way.

Reply:
I'm not really sure what the question is. The loop iterates over a list of items returned from get_sinks. Then each item can be processed and added to the gui.

Is there a better way?

Reply:
The thing is that this specific loop does not iterate as such. It breaks unconditionally after the first iteration. So I'm just curious if this has any specific meaning.

Reply:
Ah, I didn't look at the rest of the code, yes an if is probably fine here.